File: backend/CafeCRM/src/main/java/org/example/cafecrm/domain/entity/Tables.java
package org.example.cafecrm.domain.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.example.cafecrm.domain.values.TableStatus;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "restaurant_table")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Tables {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "table_id")
    private Integer id;

    @Column(nullable = false)
    private Integer number;

    @Column(nullable = false)
    private Integer capacity;

    @Column(name = "pos_x")
    private Integer posX;

    @Column(name = "pos_y")
    private Integer posY;

    @Column(nullable = false, length = 10)
    @Enumerated(EnumType.STRING)
    private TableStatus status;

//    @OneToMany(mappedBy = "table", fetch = FetchType.LAZY)
//    private List<Order> orders = new ArrayList<>();
//
    @OneToMany(mappedBy = "table", fetch = FetchType.LAZY)
    private List<Reservation> reservations = new ArrayList<>();
}

File: frontend/cafe-crm-frontend/src/types/tables.ts
export const TableStatus = {
  FREE: 'FREE',
  OCCUPIED: 'OCCUPIED',
  RESERVED: 'RESERVED',
} as const;

export type TableStatus = (typeof TableStatus)[keyof typeof TableStatus];

export interface Table {
  tableId: number;
  number: number;
  capacity: number;
  posX?: number;
  posY?: number;
  status: TableStatus;
}

export interface CreateTableDto {
  number: number;
  capacity: number;
  posX: number;
  posY: number;
}

export interface UpdateTablePositionDto {
  posX: number;
  posY: number;
}

export interface UpdateTableStatusDto {
  status: TableStatus;
}
